examples/zephyr/http-client/Makefile

```diff
@@ -1,59 +1,40 @@
-MONGOOSE_ROOT ?= $(realpath $(CURDIR)/../../..)
-DOCKER_PROJECT_DIR ?= /workdir
-ZEPHYR_DIR ?= zephyrproject
-PROJECT_NAME = $(notdir $(CURDIR))
-PROJECT_PATH = $(realpath $(CURDIR))
-ZEPHYR_PATH = $(realpath $(CURDIR)/..)/$(ZEPHYR_DIR)
-DOCKER_PROJECT_PATH = $(DOCKER_PROJECT_DIR)/$(PROJECT_NAME)
-DOCKER_ZEPHYR_PATH = $(DOCKER_PROJECT_DIR)/$(ZEPHYR_DIR)
-
-BOARD ?= qemu_x86
-ifeq "$(BOARD)" "qemu_x86"
-OVERLAY ?= -DOVERLAY_CONFIG=overlay-e1000.conf
-else
-OVERLAY ?= 
-endif
-
-SHAREUSB ?= --privileged -v /dev/bus/usb/:/dev/bus/usb
-SHARETUN ?= --cap-add=NET_ADMIN --device /dev/net/tun --net=host
-DOCKER ?= docker run --rm -v $(PROJECT_PATH):$(DOCKER_PROJECT_PATH) -v $(ZEPHYR_PATH):$(DOCKER_ZEPHYR_PATH)
-REPO ?= zephyrprojectrtos/ci
-
-
-example:
+TOP_DIR ?= $(realpath $(CURDIR)/../../..)
+ZEPHYR_DIR ?= $(TOP_DIR)/zephyrproject
+DOCKER ?= docker run --rm \
+          -v $(TOP_DIR):$(TOP_DIR) \
+          -w $(realpath $(CURDIR)) \
+          -e ZEPHYR_BASE=$(ZEPHYR_DIR)/zephyr \
+          $(DOCKER_ARGS) zephyrprojectrtos/ci
+#BUILD_ARGS ?= -b qemu_x86 -DOVERLAY_CONFIG=overlay-e1000.conf
+BUILD_ARGS ?= -b nucleo_f429zi
+
+.PHONY: build flash clean pristine run
+
+example zephyr:
 	true
 
-build:
-	cp $(MONGOOSE_ROOT)/mongoose.[c,h] src/
-	$(DOCKER) $(REPO) /bin/sh -c 'cd $(DOCKER_ZEPHYR_PATH)/zephyr && \
-	west build -b $(BOARD) -p auto $(DOCKER_PROJECT_PATH) $(OVERLAY) --build-dir $(DOCKER_PROJECT_PATH)/build'
+$(ZEPHYR_DIR):
+	mkdir -p $@
+	$(DOCKER) west init $@
+	$(DOCKER) sh -c 'cd $@ && west update'
```

Comment:
The reason I wrote it the way I did is because:
- west init has to be run only once, it takes forever, and creates the local repo
- west update has to be run after west init the first time, and every other time you want to keep up to date with their repo. It updates from all the subtrees in the Zephyr repo (Espressif, STM, etc) and also takes a fair amount of time even when just updating nothing.

If you want to get rid of the zephyr target, perhaps the best approach, at the cost of always updating the git repo, is to do as in the newer Makefiles cloning from mongoose-examples:

```
$(ZEPHYR_DIR):
	mkdir
	west init

build: | $(ZEPHYR_DIR)
	west update
```
The pipe is supposed to just check if the directory exists so west init will only be executed when the folder is missing

However, with properly written tests, there will be a noticeable speed up if only the first Zephyr example clones the repo and the others just use it and don't try to re-update it.

Reply:
I believe this approach implements this idea. The `zephyrproject` dir is created in the mongoose repo root, and can be reused by other examples.

I think running `west update` on every build is a overkill; it executes extremely slowly on my machine (takes minutes)

Reply:
Restored clean and pristine targets. Added `purge` target which deletes everything